models/iaf_cond_alpha_mc_kinetics.cpp

```diff
@@ -0,0 +1,875 @@
+  // will optimized most stuff away ...
+  for ( size_t n = 0; n < N::NCOMP; ++n )
+  {
+    // membrane potential for current compartment
+    const double V = y[ S::idx( n, S::V_M ) ];
+
+    // excitatory synaptic current
+    const double I_syn_exc = ( y[ S::idx( n, S::G_EXC ) ] ) * ( V - node.P_.E_ex[ n ] );
+    // inhibitory synaptic current
+    const double I_syn_inh = ( y[ S::idx( n, S::G_INH ) ] ) * ( V - node.P_.E_in[ n ] );
+
+    // leak current
+    const double I_L = node.P_.g_L[ n ] * ( V - node.P_.E_L[ n ] );
+
+    // changes
+
+    double minf = 0.0;
+    double hinf = 0.0;
+    if ( n == N::DIST )
+    {
+      if ( node.P_.act_flag == 1.0 )
+      {
+        minf =
+          1.0 / ( 1.0 + pow( numerics::e, ( ( V - node.P_.half_m ) * -1.0 * node.P_.slope_m ) ) );
+        hinf =
+          1.0 / ( 1.0 + pow( numerics::e, ( ( V - node.P_.half_h ) * -1.0 * node.P_.slope_h ) ) );
+      }
+    }
+
+
+    const double ica =
+      y[ S::idx( n, S::M_CA ) ] * y[ S::idx( n, S::H_CA ) ] * node.P_.G_Ca * ( node.P_.E_Ca - V );
+
+    // coupling currents
+    const double I_conn =
+      ( n > N::SOMA
+          ? node.P_.g_conn[ n - 1 ]
+            * ( ( V - node.P_.E_L[ n ] ) - ( y[ S::idx( n - 1, S::V_M ) ] - node.P_.E_L[ n - 1 ] ) )
+          : 0 )
+      + ( n < N::NCOMP - 1
+            ? node.P_.g_conn[ n ] * ( ( V - node.P_.E_L[ n ] )
+                                      - ( y[ S::idx( n + 1, S::V_M ) ] - node.P_.E_L[ n + 1 ] ) )
+            : 0 );
+
+    // derivatives
+    // membrane potential
+    f[ S::idx( n, S::V_M ) ] =
+      ( -I_L - I_syn_exc - I_syn_inh - I_conn + node.B_.I_stim_[ n ] + node.P_.I_e[ n ] + ica
+        + y[ S::idx( n, S::I_AP ) ] ) / node.P_.C_m[ n ];
+
+    // excitatory conductance
+    f[ S::idx( n, S::DG_EXC ) ] = -y[ S::idx( n, S::DG_EXC ) ] / node.P_.tau_synE[ n ];
+    f[ S::idx( n, S::G_EXC ) ] =
+      y[ S::idx( n, S::DG_EXC ) ] - y[ S::idx( n, S::G_EXC ) ] / node.P_.tau_synE[ n ];
+
+    // inhibitory conductance
+    f[ S::idx( n, S::DG_INH ) ] = -y[ S::idx( n, S::DG_INH ) ] / node.P_.tau_synI[ n ];
+    f[ S::idx( n, S::G_INH ) ] =
+      y[ S::idx( n, S::DG_INH ) ] - y[ S::idx( n, S::G_INH ) ] / node.P_.tau_synI[ n ];
+
+    // active current during AP
+    f[ S::idx( n, S::DI_AP ) ] = -y[ S::idx( n, S::DI_AP ) ] / node.P_.tau_curAP[ n ];
+    f[ S::idx( n, S::I_AP ) ] =
+      y[ S::idx( n, S::DI_AP ) ] - y[ S::idx( n, S::I_AP ) ] / node.P_.tau_curAP[ n ];
+
+    if ( n == N::DIST )
+    {
+      if ( node.P_.act_flag == 1.0 )
+      {
+        f[ S::idx( n, S::M_CA ) ] = ( minf - y[ S::idx( n, S::M_CA ) ] ) / node.P_.tau_m;
+        f[ S::idx( n, S::H_CA ) ] = ( hinf - y[ S::idx( n, S::H_CA ) ] ) / node.P_.tau_h;
+      }
+      else
+      {
+        f[ S::idx( n, S::M_CA ) ] = 0.0;
+        f[ S::idx( n, S::H_CA ) ] = 0.0;
+      }
+    }
+    else
+    {
+      f[ S::idx( n, S::M_CA ) ] = 0.0;
+      f[ S::idx( n, S::H_CA ) ] = 0.0;
+    }
```

Comment:
Can you change the logic so that you do not need nested if-else with two identical else clauses?